zip-0228.rst

```diff
@@ -2,12 +2,530 @@
+This may pose regulatory issues for the Matcher (e.g. if specific assets are deemed illegal in specific jurisdictions). 
+The effect of this may be the re-creation of mechanisms of central exchanges, where only a specific list of assets are accepted for trading by matchers (similar to the “asset listing” approach of centralized exchanges).
+
+
+Matchers
+--------
+
+The Swap protocol has been described above for the peer-to-peer setting between parties. However, it also supports other use cases such as transaction relays and centralized matching services.
+This is achieved via Matchers, who are entities (or a network of entities) that run an Order Matching process. This Matching process (i.e. instance of computer program) carries out the Matching operation on a set of received Orders. The Matcher produces Bundles from Orders and then sends them on-chain for settlement. Some supported possibilities are sketched in the figures below.
+
+.. figure:: zip-0228-images/zip-0228-matcher-scenario-1.png
+    :width: 600px
+    :align: center
+    :figclass: align-center
+
+    Scenario 1: The order matching process is run on a single machine (i.e the Matcher is one machine used by User A and User B to swap).
+
+.. figure:: zip-0228-images/zip-0228-matcher-scenario-2.png
+    :width: 600px
+    :align: center
+    :figclass: align-center
+
+    Scenario 2: The order matching process is run via multiple machines (they could be connected on a network).
+
+High-level logic for Order Matching
+```````````````````````````````````
+
+The Order Matching is an off-chain procedure, and therefore is out of scope for this ZIP. The logic followed will be Matcher-dependent, but we assume that it follows the three high-level steps described below. 
+We also assume the Matcher gets paid a fee (paid in ZEC) for its service, and that Orders are matched perfectly (that is, there is no partial fulfilment of a Swap Order).
+
+Receive the Swap Order and Perform Order Matching
+'''''''''''''''''''''''''''''''''''''''''''''''''
+
+The Matcher must be able to receive incoming Swap Orders from traders willing to use it to Swap ZSAs. We assume that a Matcher is discoverable by market participant (e.g. it has a website), that traders can send Orders to the Matcher (e.g. using a secure 1-to-1 communication channel) and that a Matcher defines its own logic to process the received Swap Orders. Examples of such logic are:
+
+- A set of validity checks to run on the received Swap Orders, e.g. to prevent DoS attacks (See `Denial-of-Service (DoS) Prevention`_). If the verification of a Swap Order fails, the Swap Order gets rejected.
+- A set of operations to manage the valid Swap Orders waiting to be matched. This could include, for instance, a set of data structures and operations to manage a mempool/Central Limit Order Book (CLOB), following a chosen ordering policy (See, `CLOB and Matching Policies`_ for more details), etc.
```

Comment:
Nitpick: CLOB is used earlier in the text, yet, the acronym is unrolled here. Could be worth unrolling the acronym only once, the first time CLOB is mentioned in the ZIP.

Reply:
Unrolled only once at the start, switched to CLOB in later uses. In #62